helm-charts/auth-apisix/README.md

```diff
@@ -0,0 +1,113 @@
+
+```sh
+# Prerequisite: Create a PersistentVolume of 9Gi for keycloak-postgress with RWO access (to persist updated keycloak configuration)
+# Below is a reference to create PersistentVolume which will be used by keycloak-postgress
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: data-keycloak-postgresql-0
+spec:
+  capacity:
+    storage: 9Gi
+  hostPath:
+    path: "/mnt/keycloak-vol"
+  accessModes:
+    - ReadWriteOnce
+EOF
+
+# install keycloak with helm by setting
+helm install keycloak oci://registry-1.docker.io/bitnamicharts/keycloak --version 22.1.0 --set auth.adminUser=admin --set auth.adminPassword=admin
+
+# Access keycloak UI through service/keycloak to do the necessary configurations
+```
+
+Once the keycloak pod is up and running, access the UI through the Keycloak's NodePort service and set the necessary configuration
+
+## Update values
+
+Update the following values in values.yaml
+
+1. Update all the entries in oidc config pertaining to your identity provider
+
+2. Update all the entries in API specific configs
+
+## Install
+
+```sh
+# Install apisix api gateway and ingress controller
+helm install auth-apisix apisix/apisix -f values_apisix_gw.yaml --create-namespace --namespace auth-apisix
+
+# WAIT UNTIL apisix-ingress-controller POD IS READY by checking status with 'kubectl get -n auth-apisix pods'
+# The pod is ready when READY status shows 1/1
+
+# Publish authenticated APIs in APISIX gateway
+helm install auth-apisix-crds . --namespace auth-apisix
+```
+
+## Usage
+
+The published APIs in apisix gateway are accessible through auth-apisix-gateway kubernetes service. By default, it is a NodePort service and can be accessed as:
+
+```sh
+export NODE_PORT=$(kubectl get --namespace auth-apisix -o jsonpath="{.spec.ports[0].nodePort}" services auth-apisix-gateway)
+export NODE_IP=$(kubectl get nodes --namespace auth-apisix -o jsonpath="{.items[0].status.addresses[0].address}")
+
+# the authenticated endpoint published in APISIX gateway can be accessed as: http://$NODE_IP:$NODE_PORT/<published endpoint uri>
+export accessUrl=http://$NODE_IP:$NODE_PORT/<your published endpoint uri>
+
+
+```
+
+</br>
+Apisix helm chart provides configs to change the service type to other options like LoadBalancer (apisix.service.type) and externalTrafficPolicy to 'local'(apisix.service.externalTrafficPolicy). These can be added in values_apisix_gw.yaml </br></br>
+While accessing the published APIs, the HTTP Authorization header of the request should contain the Access token provided by Identity provider as "Bearer &lt;Access Token&gt;" </br></br>
+The access token, refresh token, userinfo and user roles can be obtained by invoking OIDC auth endpoint through UI or token endpoint through curl and providing user credentials. </br></br>
+
+Below steps can be followed to get access token from keycloak and access the APISIX published ChatQnA API through curl
+
```

Comment:
Please help explain your authentication and authorization scenarios here.

Reply:
I agree the proposal of the above folder layout.